### DISCUSSION TITLE
Single agent reporting to more than one server?

### DISCUSSION THREAD

**Discussion by yurividal** [henrygd/beszel#341] [Support]
Hi. i have 2 beszel servers running. One on-site and one off-site. 
Both areas are connected via a site-to-site VPN, and reachable by IP.
Is it possible to have my beszel agent on one of my servers report to both beszels (offsite and onsite) without having to run 2 separate docker containers? (It's an old raspbery pi, so running 2 agents seems like a waste of the limited resources.

**Comment by henrygd** [comment]
The expectation with the current design is that an agent will communicate with one hub.

Considering your scenario, I agree that one agent connecting to multiple hubs would be ideal. I'll try to add support for this in the next release. I'm visiting family right now so it may be a few weeks.

**Comment by yurividal** [comment]
Hey @henrygd , just wondering if you had a chance to look into this?

**Reply by henrygd** [reply]
I was just thinking about this yesterday. I'll do a quick test tomorrow and try to squeeze it into the next release if that goes well.

**Comment by eduardolat** [comment]
This would be really helpful.

In my case, I have a few servers for my clients (I'm a freelancer) and I would like the agent to report to the individual hubs but I also want to have a global hub for all servers.

Thanks for making this project so cool and easy to use

**Comment by henrygd** [comment]
This has been added. New release will be out later today if I have time to update the docs and release notes, otherwise tomorrow.

**Reply by eduardolat** [reply]
That sounds great! Thank you so much for this great tool

**Comment by henrygd** [comment]
This should work now in 0.10.0.

Separate your keys with a new line using `\n` or a multiline string. The new `KEY_FILE` env var should also work.

Let me know if you have any trouble with it!

**Reply by wintech147** [reply]
Hi @henrygd I was able to use two keys on one agent via the docker compose file generated and adding the other key but I'm new to linux and not sure how to do that for the binary install option where it generates the curl command. Can you provide an example? Please and thank you!

**Reply by henrygd** [reply]
See this section in the docs and let me know if that doesn't get you there. You would probably be using Systemd.

https://beszel.dev/guide/environment-variables#binary

**Reply by wintech147** [reply]
Worked like a charm. Thank you for the quick reply and amazing work!

**Reply by henrygd** [reply]
No problem :+1:

**Reply by wintech147** [reply]
Sent some coffee! Enjoy! <3

**Reply by henrygd** [reply]
Gracias

**Comment by eduardolat** [comment]
It worked perfectly fine for me! Thank you very much for this excellent work.

It would be great to add an example of this in beszel.dev, it would also be good to document that the file in KEY_FILE supports blank lines and comments with # (I have read the code)

It can be very useful for everyone

![image](https://github.com/user-attachments/assets/2d5a4045-b991-4a0a-9167-19c87e454ef8)
